template/koa-rest-api/src/framework/validation.ts

```diff
@@ -35,41 +35,38 @@ type InvalidFields = Record<string, string>;
  * @see [union error example](./validation.test.ts)
  */
 const parseInvalidFieldsFromError = (err: z.ZodError): InvalidFields =>
-  Object.fromEntries(parseTuples(err, {}));
+  Object.fromEntries(parseTuples(err.issues));
 
 const parseTuples = (
-  { errors }: z.ZodError,
-  unions: Record<number, number[]>,
+  errors: core.$ZodIssue[],
+  basePath: Array<string | number | symbol> = [],
+  unions: Record<number, number[]> = {},
 ): Array<readonly [string, string]> =>
   errors.flatMap((issue) => {
-    if (issue.code === ZodIssueCode.invalid_union) {
-      return issue.unionErrors.flatMap((err, idx) =>
-        parseTuples(err, {
+    if (issue.code === 'invalid_union') {
+      return issue.errors.flatMap((err, idx) =>
+        parseTuples(err, issue.path, {
```

Comment:
The base path doesn't get added to the children anymore so we need to carry it forward